docs/using/frameworks/django.rst

```diff
@@ -0,0 +1,166 @@
+======
+Django
+======
+
+The oso Django integration adopts Django conventions and provides middleware,
+view decorators and ORM integrations to make it easier to use oso with Django.
+
+Installation
+============
+
+The oso Django integration is available on `PyPI`_ and can be installed using
+``pip``::
+
+    $ pip install django-oso
+
+.. _PyPI: https://pypi.org/project/django-oso/
+
+Usage
+=====
+
+The ``django_oso`` django plugin contains a reusable django app that makes
+authorization with oso and django easy.  To use, ensure ``django_oso`` is in
+``INSTALLED_APPS``:
+
+.. code-block:: python
+    :caption: :fab:`python` settings.py
+
+    INSTALLED_APPS = [
+        'django_oso',
+        ...
+    ]
+
+Loading policies
+----------------
+
+``django_oso`` expects policy files to be included in the ``policy`` directory
+of each installed app.  Upon startup, all ``.polar`` files found in that
+directory (or sub-directories) will be loaded using
+:py:meth:`oso.Oso.load_file`.  To load additional files outside of these
+directories, call :py:meth:`~oso.Oso.load_file` on
+:py:data:`django_oso.oso.Oso`.
+
+Registering classes & models
+----------------------------
+
+Often, authorization rules will be expressed over django models.  Therefore,
+``django_oso`` will register every model for each installed app upon startup as
+a class with oso. The :py:class:`django.http.HttpRequest` is also registered
+under ``HttpRequest``.  Django models are referenced in a Polar file using the
+syntax ``app_name::ModelName``. If an app name contains `.`, for example
+``django.contrib.auth``, it will be referenced in oso as
+``django::contrib::auth``.
+
+Additional classes can be registered as needed using
+:py:meth:`oso.Oso.register_class` on :py:data:`django_oso.oso.Oso`.
+
+Performing authorization
+------------------------
+
+To authorize a request, use the :py:func:`django_oso.auth.authorize` function.
+It accepts the same arguments as
+:py:meth:`~oso.Oso.is_allowed`, but provides sensible defaults for working with
+Django. The actor defaults to ``request.user``.  The ``action``
+defaults to the method of the request.
+``resource`` must be provided.
+
+.. tip::
+
+    If you aren't familiar with how oso uses actors, actions, and resources to
+    express authorization decisions, see :doc:`/more/glossary` or
+    :doc:`/getting-started/quickstart`.
+
+:py:meth:`django_oso.auth.authorize` can be used within route handlers, or in
+the data access layer, depending upon how you want to express authorization.
+
+Here's a basic example in a route:
+
+
+.. code-block:: python
+    :emphasize-lines: 7
+
+    def get_expense(request, id):
+        try:
+            expense = Expense.objects.get(pk=id)
+        except Expense.DoesNotExist:
+            return HttpResponseNotFound()
+
+        authorize(request, expense, action="read")
+        return HttpResponse(expense.json())
+
+Requiring authorization on every request
+----------------------------------------
```

Comment:
Something that we shouldn't address on this PR, but it might be worth talking about how using this middleware (and the one in the Flask integration) will only effectively apply authorization if there is a corresponding rule in your policy for the actor, action, and resource.